src/librados/IoCtxImpl.cc

```diff
@@ -742,6 +742,11 @@ int librados::IoCtxImpl::operate_read(const object_t& oid,
 
   Context *onack = new C_SafeCond(&mylock, &cond, &done, &r);
 
+  // invalidate any existing crc (if the buffer is reused)
+  if (pbl) {
```

Comment:
would aio_operate_read() have the same problem?

Reply:
Looking into this now.